x/gamm/types/msgs_test.go

```diff
@@ -523,23 +523,23 @@ func TestMsgJoinSwapExternAmountIn(t *testing.T) {
 		{
 			name: "invalid denom",
 			msg: createMsg(func(msg gammtypes.MsgJoinSwapExternAmountIn) gammtypes.MsgJoinSwapExternAmountIn {
-				msg.TokenIn.Denom = "1"
+				msg.TokensIn[0].Denom = "1"
 				return msg
 			}),
 			expectPass: false,
 		},
 		{
 			name: "zero amount",
 			msg: createMsg(func(msg gammtypes.MsgJoinSwapExternAmountIn) gammtypes.MsgJoinSwapExternAmountIn {
-				msg.TokenIn.Amount = sdk.NewInt(0)
+				msg.TokensIn[0].Amount = sdk.NewInt(0)
 				return msg
 			}),
 			expectPass: false,
 		},
 		{
 			name: "negative amount",
 			msg: createMsg(func(msg gammtypes.MsgJoinSwapExternAmountIn) gammtypes.MsgJoinSwapExternAmountIn {
-				msg.TokenIn.Amount = sdk.NewInt(-10)
+				msg.TokensIn[0].Amount = sdk.NewInt(-10)
```

Comment:
Can we add the two token test to `TestMsgJoinSwapExternAmountIn` instead?

`TestAuthZ`'s goal is slightly different